v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/schema/SessionBasedMapper.java

```diff
@@ -149,4 +155,22 @@ public List<String> getSpannerColumns(String namespace, String spannerTable)
       throws NoSuchElementException {
     return schema.getSpannerColumnNames(spannerTable);
   }
+
+  @Override
+  public String getShardIdColumnName(String namespace, String spannerTableName) {
+    Map<String, NameAndCols> spanToId = schema.getSpannerToID();
+    if (!spanToId.containsKey(spannerTableName)) {
+      throw new NoSuchElementException(
+          String.format("Spanner table '%s' not found", spannerTableName));
+    }
+    try {
+      String tableId = spanToId.get(spannerTableName).getName();
+      SpannerTable table = schema.getSpSchema().get(tableId);
+      String colId = table.getShardIdColumn();
```

Comment:
If colId is null, or does not exist in col definition, we should throw an explicit exception. Catching Null pointers and throwing a runtime exception is not really a good idea. Lets used types exceptions here.

Reply:
Updated